tests/httpfake.h

```diff
@@ -14,22 +14,35 @@
 #include "crypto/crypto.h"
 #include "http/httpinterface.h"
 #include "logging/logging.h"
+#include "metafake.h"
 #include "utilities/utils.h"
 
 enum class ProvisioningResult { kOK, kFailure };
 
 class HttpFake : public HttpInterface {
  public:
   // old style HttpFake with centralized multi repo and url rewriting
-  HttpFake(const boost::filesystem::path &test_dir_in, std::string flavor = "",
-           const boost::filesystem::path &meta_dir_in = "")
+  HttpFake(const boost::filesystem::path &test_dir_in,
+           std::string flavor = "",
+           const boost::filesystem::path &meta_dir_in = "",
+           const boost::filesystem::path &meta_orig_in = "")
       : test_dir(test_dir_in), flavor_(std::move(flavor)), meta_dir(meta_dir_in) {
     if (meta_dir.empty()) {
       meta_dir = temp_meta_dir.Path();
-      Utils::copyDir("tests/test_data/repo/repo/image", meta_dir / "repo");
-      Utils::copyDir("tests/test_data/repo/repo/director", meta_dir / "director");
-      if (boost::filesystem::is_directory("tests/test_data/repo/campaigner")) {
-        Utils::copyDir("tests/test_data/repo/campaigner", meta_dir / "campaigner");
+      boost::filesystem::path fake_meta_path;
+
+      if (meta_orig_in.empty()) {
```

Comment:
Hm, unfortunately it seems to not help much for clarity. Having meta_dir_in and meta_orig_in is confusing and it's worsened by the fact that a lot of the `HttpFake` constructions now give an empty `""` in the middle of the arguments.

A big part of this mess predates this PR though, legacy of all the different ways we had to use metadata in tests.

I would expect that the directory generated from `MetaFake` could be directly be passed in `meta_dir_in`.

As it would be great to get rid of these: https://github.com/advancedtelematic/aktualizr/blob/12f8558788dd23b6ce36fedb5c3517822a865a66/src/libaktualizr/primary/aktualizr_fullostree_test.cc#L159 (there are 4 instances in the code base), maybe it's worth changing this line https://github.com/advancedtelematic/aktualizr/blob/12f8558788dd23b6ce36fedb5c3517822a865a66/tests/uptane_test_common.h#L42 to map to what aktualizr-repo spits out.

Then hopefully, we can simplify all of this.